packages/core/src/Dialog/Dialog.styled.tsx

```diff
@@ -254,7 +255,7 @@ export const DialogContent = ({
               )}
             </SmoothTransitionBox>
           ) : (
-            <Box height='100%' style={{ overflowY: 'scroll' }}>
+            <Box height='100%' style={innerContentScroll ? { overflowY: 'scroll' } : {}}>
```

Comment:
```suggestion
            <Box height='100%' style={innerContentScroll ? { overflowY: 'scroll' } : undefined}>
```

Reply:
There is a prop `showScrollShadow`. When the value for it is `true`, this `Box` is not shown. Instead, the `SmoothTransitionBox` is shown above. It seems that will have the same problem. So we need to tackle this on all fronts